src/components/Navigation/Navigation.js

```diff
@@ -0,0 +1,108 @@
+import {useState} from 'react';
+import {IconContext} from 'react-icons';
+import * as FaIcons from 'react-icons/fa';
+import * as MdIcons from 'react-icons/md';
+import {Link} from 'react-router-dom';
+import styled from 'styled-components';
+
+import {NavigationData} from './NavigationData';
+
+export default function Navigation() {
+  const [showSidebar, setShowSidebar] = useState(false);
+
+  const toggleSidebar = () => setShowSidebar(!showSidebar);
+  return (
+    <>
+      <IconContext.Provider value={{color: '#FFF'}}>
+        <MenuDiv>
+          <MenuOpenButton aria-label="Open menu" onClick={toggleSidebar}>
+            <FaIcons.FaBars />
+          </MenuOpenButton>
+        </MenuDiv>
+        <NavMenu isOpen={showSidebar}>
+          <MenuList onClick={toggleSidebar}>
+            <MenuItem>
+              <MenuDiv>
+                <MenuCloseButton aria-label="Open menu" onClick={toggleSidebar}>
+                  <MdIcons.MdClose />
+                </MenuCloseButton>
+              </MenuDiv>
+            </MenuItem>
+            {NavigationData.map((item, index) => {
+              return (
+                <li key={index}>
+                  <Link to={item.path}>
+                    {item.icon}
+                    <span>{item.title}</span>
+                  </Link>
+                </li>
+              );
+            })}
+          </MenuList>
+        </NavMenu>
+      </IconContext.Provider>
+    </>
+  );
```

Comment:
Regarding your comment on the navbar not closing. What might be a good way to do this then, would be to use buttons instead of links and the `useNavigate` hook from react router: https://reactrouter.com/docs/en/v6/hooks/use-navigate

It could look like this:
```suggestion
  const [showSidebar, setShowSidebar] = useState(false);
  const navigate = useNavigate();
  
  const navigateToPage = (path) => {
    navigate(path)
    setShowSidebar(false)
  }

  const toggleSidebar = () => setShowSidebar(!showSidebar);
  return (
    <>
      <IconContext.Provider value={{color: '#FFF'}}>
        <MenuDiv>
          <MenuOpenButton aria-label="Open menu" onClick={toggleSidebar}>
            <FaIcons.FaBars />
          </MenuOpenButton>
        </MenuDiv>
        <NavMenu isOpen={showSidebar}>
          <MenuList>
            <MenuItem>
              <MenuDiv>
                <MenuCloseButton aria-label="Open menu" onClick={toggleSidebar}>
                  <MdIcons.MdClose />
                </MenuCloseButton>
              </MenuDiv>
            </MenuItem>
            {NavigationData.map((item, index) => {
              return (
                <li key={index}>
                  <button onClick={() => navigateToPage(item.path)}>
                    {item.icon}
                    <span>{item.title}</span>
                  </button>
                </li>
              );
            })}
          </MenuList>
        </NavMenu>
      </IconContext.Provider>
    </>
  );
```
That way the semantics would be correct and the `onClick` would be on a button. However, I didn't try this and it is just my personal preference/suggestion to do this. :)

Reply:
Thank you! Added the hook, button, labels and done the styling. 👍

Reply:
Nice job ! :)